include/courtier/GIoContexts.hpp

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <cstdlib>
+#include <climits>
+#include <thread>
+#include <pthread.h>
+#include <vector>
+#include <mutex>
+
+#include <boost/version.hpp>
+#include <boost/asio.hpp>
+#include <boost/noncopyable.hpp>
+#include <boost/shared_ptr.hpp>
+
+#include "common/GGlobalDefines.hpp"
+#include "common/GLogger.hpp"
+#include "common/GErrorStreamer.hpp"
+
+
+namespace Gem {
+  namespace Courtier {
+
+    namespace net = boost::asio;
+    
+    class GIoContexts
+      : private boost::noncopyable
+    {
+    public:
+      
+      GIoContexts(int c_size)
```

Comment:
Suggest to add a small doxygen description to each member function, describing purpose and arguments